src/PrestaShopBundle/Resources/config/services/core/grid/grid_data_factory.yml

```diff
@@ -687,3 +687,21 @@ services:
       - '@prestashop.core.hook.dispatcher'
       - '@prestashop.core.grid.query.doctrine_query_parser'
       - 'image_type'
+
+  prestashop.core.grid.data_factory.hook:
+    class: '%prestashop.core.grid.data.factory.doctrine_grid_data_factory%'
+    public: true
+    arguments:
+      - '@prestashop.core.api.query_builder.hook'
+      - '@prestashop.core.hook.dispatcher'
+      - '@prestashop.core.grid.query.doctrine_query_parser'
+      - 'hook'
+
+  prestashop.core.grid.data_factory.module:
+    class: '%prestashop.core.grid.data.factory.doctrine_grid_data_factory%'
+    public: true
+    arguments:
+      - '@prestashop.module.api.list.module_query_builder'
+      - '@prestashop.core.hook.dispatcher'
+      - '@prestashop.core.grid.query.doctrine_query_parser'
+      - 'module'
```

Comment:
Same remark, this one should be in the module since the query builder is implemented and defined in the module We can't have dependencies on services from modules in the core because modules can be uninstalled and their related services will disappear thus creating DI errors